functional_testing/Openvcloud/RESTful/testcases/libcloud/test01_libvirt.py

```diff
@@ -0,0 +1,159 @@
+        
+    def test01_get_free_mac_address(self):
+        """ OVC-001
+        #. Get free mac address, should succeed.
+        #. Get free mac address of invalid grid id, should fail.
+        """
+        self.lg.info('Get free mac address, should succeed')
+        response = self.api.libcloud.libvirt.getFreeMacAddress(gid=self.gid)
+        self.assertEqual(response.status_code, 200)
+        self.assertTrue(response.text)
+
+        self.lg.info('Get free mac address of invalid grid id, should fail')
+        gid = self.utils.random_string()
+        response = self.api.libcloud.libvirt.getFreeMacAddress(gid=gid)
+        self.assertEqual(response.status_code, 400)
+                
+    def test02_get_free_network_id(self):
+        """ OVC-002
+        #. Get free network id, should succeed.
+        #. Get free network id of invalid grid id, should fail.
+        """
+        self.lg.info('Get free network id, should succeed')
+        response = self.api.libcloud.libvirt.getFreeNetworkId(gid=self.gid)
+        self.assertEqual(response.status_code, 200)
+        self.assertTrue(response.text)
+
+        self.lg.info('Get free network id of invalid grid id, should fail')
+        gid = self.utils.random_string()
+        response = self.api.libcloud.libvirt.getFreeNetworkId(gid=gid)
+        self.assertEqual(response.status_code, 400)
+
+    def test03_register_release_network_id_range(self):
+        """ OVC-003
+        #. Create account (AC1), should succeed.
+        #. Create cloudspace (CS1), should succeed.
+        #. Register network id, should succeed.
+        #. Register network id in range of deployed networkids, should fail
+        #. Register network id of invalid grid id, should fail.
+        #. Register network id of invalid start value, should fail.
+        """
+        self.lg.info('Create account (AC1), should succeed')
+        data, response = self.api.cloudbroker.account.create(username=self.whoami)
+        self.assertEqual(response.status_code, 200, response.content)
+        account_id = int(response.text)
+        self.CLEANUP['accounts'].append(account_id)
+
+        self.lg.info('Create cloudspace (CS1), should succeed')
+        data, response = self.api.cloudapi.cloudspaces.create(accountId=account_id, location=self.location, access=self.whoami)
+        self.assertEqual(response.status_code, 200)
+        cloudspace_id = int(response.text)
+
+        for _ in range(20):
+            response = self.api.cloudapi.cloudspaces.get(cloudspaceId=cloudspace_id)
+            if response.json()['status'] == 'DEPLOYED':
+                break
+            time.sleep(5)
+        else:
+            self.fail('Can\'t create cloudspace')
+
+        response = self.api.cloudbroker.cloudspace.getVFW(cloudspaceId=cloudspace_id)
+        self.assertEqual(response.status_code, 200)
+        network_id = int(response.json()['id'])
+        
+        self.lg.info('Register network id, should succeed')
```

Comment:
from here to the end of the method is a duplicate code. I suggest to use parametrized instead of repeat code lines